scripts/setup/get-icons.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const { exec } = require('child_process');
+const fs = require('fs');
+const chalk = require('chalk');
+
+const callback = (error, stdout, stderr) => {
+  if (error) {
+    console.error(chalk`{red exec error: ${error}}`);
+    return;
+  }
+
+  if (stderr) {
+    console.error(`stderr: ${stderr}`);
+    return;
+  }
+
+  console.log(`stdout: ${stdout}`);
+  console.log(chalk`{green The icons have been successfully obtained. Check public/assets/icons/}`);
+};
+
+let rawdata = fs.readFileSync('./.resources.json');
+let data = JSON.parse(rawdata);
+
+if (data.hasOwnProperty('url')) {
+  exec('[ -d "public/assets/icons/" ] && rm -rf public/assets/icons');
+  exec('mkdir -p public/assets/icons');
+  exec(`svn checkout ${data.url} public/assets/icons/`, (error, stdout, stderr) =>
```

Comment:
In our last meeting, I thought we agreed that the resources should be in a git repo. Each particular project on a different repo. E.g.:
rmf_demos/rmf_dashboard_resources/
rmf_implementationA/rmf_dashboard_resources/

But actually, with the current structure, we can point at any repo folder we want.
Also, I added a functionality for copying the resources from a specific folder inside the server in case necessary

Is it ok? Am I missing something? 

Thinking that I use svn to extract files from a specific folder inside the repo.

. SVN Checkout – Create working copy
Checkout command is used to download sources from SVN repository to working copy. If you want to access files from the SVN server, checkout is the first operation you should perform.

`SVN checkout` creates the working copy, from where you can do edit, delete, or add contents. You can checkout a file, directory, trunk or whole project. To checkout you should know URL of the components you want to checkout.

src: https://www.javatpoint.com/svn-commands and https://www.thegeekstuff.com/2011/04/svn-command-examples/#:~:text=SVN%20stands%20for%20Subversion.,placed%20into%20a%20central%20repository.&text=This%20article%20explains%20some%20basic%20SVN%20commands%20with%20examples.

Reply:
I didn't know you can checkout a git repo with svn, but I would still prefer to avoid bringing it in as a dependency.

Reply:
Ok! I'll find a way to do it without svn. As a last resort, we could create a repo (let's say `rmf-dashboard-resources`) and add it as a git sub-module to the `rmf-demos` project and make a direct clone in this project (`romi-dashboard`)

Reply:
Update. I found this information:
* GH does not support git-archive https://stackoverflow.com/questions/7106012/download-a-single-folder-or-directory-from-a-github-repo#:~:text=Step1%3A%20Input%20github%20url%20to,File%22%20button%20to%20get%20files.
* Using the oldest version of git sparse-checkout will still download the whole repository from the server – only the checkout is reduced in size. At the moment it is not possible to clone only a single directory. But if you don't need the history of the repository, you can at least save on bandwidth by creating a shallow clone. 

* The newest (better I found discarding svn option) is to use the new version of sparse-checkout, with no checkout and with no history. Note that it requires git version 2.25 installed. Read more about it here: https://github.blog/2020-01-17-bring-your-monorepo-down-to-size-with-sparse-checkout/

Reply:
The `git sparse-checkout` was implemented. What do you think about it?